client/src/push-content.ts

```diff
@@ -84,10 +86,29 @@ export const getMessage = async (): Promise<string | undefined> => {
   return message
 }
 
-export const pushContent = () => async () => {
-  if (await canPush(getErrorDiagnosticsBySource())) {
-    await _pushContent(getRepo, getMessage, vscode.window.showInformationMessage, vscode.window.showErrorMessage)()
-  }
+export const pushContent = (hostContext: ExtensionHostContext) => async () => {
+  await vscode.window.withProgress({
+    location: vscode.ProgressLocation.Notification,
+    title: 'Push Content',
+    cancellable: true
+  }, async (progress, token) => {
+    token.onCancellationRequested(() => {
+      console.log('User canceled the push operation')
+    })
+    // indeterminate progress https://github.com/Microsoft/vscode/issues/47387#issuecomment-379537556
+    progress.report({ message: 'Creating Auto Element IDs, please wait...' })
+
+    // const serverErrorMessage = 'Server cannot properly find workspace'
+    const uri = expect(getRootPathUri(), 'No root path in which to generate a module')
+    // fix ids
+    // TODO: better ui in future. Add `increment` value in `progress.report` and use a callback to update real progress
+    await requestEnsureIds(hostContext.client, { workspaceUri: uri.toString() })
```

Comment:
Would this start the push process and ID generation even if `canPush` returns `false`? If so, would it seem odd to inject IDs and then throw a dialogue that there are validation error?

Reply:
you're right. I will fix that before I'm gone today...